sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileStatusCache.scala

```diff
@@ -36,6 +37,8 @@ import org.apache.spark.util.SizeEstimator
 object FileStatusCache {
   private var sharedCache: SharedInMemoryCache = _
```

Comment:
I thought this is a global LRU cache, how does it cause OOM?

Reply:
@cloud-fan In fact, if we set the value of `spark.sql.hive.filesourcePartitionFileCacheSize` about the same as or even larger than the value of `spark.driver.memory`, it may cause OOM.
Even if OOM doesn't occur, we should still clean these `FileStatus` objects in the `Cache` when we close a sparkSession, because they are useless.